### ISSUE TITLE
question about first load

### ISSUE THREAD

**Issue by stewwan** [scullyio/scully#159]
this is an issue I'm trying to get rid since I tried angular universal a few years ago.

is there a way to fix these flickering/flashes on first load?

![Screen-Recording-2020-01-07-at-11 09 33](https://user-images.githubusercontent.com/719763/71901627-21eb7080-313f-11ea-9f67-822c64e1fbd9.gif)

**Comment by chrisguttandin** [comment]
Hi @stewwan, I guess you can get rid of the flickering by setting the `initialNavigation` to `'enabled'`. It's mentioned in the [Angular docs](https://angular.io/api/router/ExtraOptions#initialNavigation). It's also done by the new schematics for Angular Universal.

**Comment by stewwan** [comment]
> Hi @stewwan, I guess you can get rid of the flickering by setting the `initialNavigation` to `'enabled'`. It's mentioned in the [Angular docs](https://angular.io/api/router/ExtraOptions#initialNavigation). It's also done by the new schematics for Angular Universal.

yeah I tried that. also preloadingStrategy and even the preboot module. none of them worked =\

**Comment by osnoser1** [comment]
The flicker usually happens if you have in the developer tools the cache disabled. 
That is, enabling the cache, and using the `initialNavigation: 'enabled'` flag you should not have problems.

Now, this specific problem is because it is necessary to implement a cache strategy in http requests using the `TransferState` to serialize the requests in the html, but from what I saw from the scully render the part that the server side should do, is not yet implemented.

**Comment by stewwan** [comment]
> The flicker usually happens if you have in the developer tools the cache disabled.
> That is, enabling the cache, and using the `initialNavigation: 'enabled'` flag you should not have problems.
> 
> Now, this specific problem is because it is necessary to implement a cache strategy in http requests using the `TransferState` to serialize the responses in the html, but from what I saw from the scully render the part that the server side should do, is not yet implemented.

even with initialNavigation and cache enabled.
flash is still there

check it out

https://scully-demo.stewan.io/
https://next.stewan.io/

but I agree about the transferState, that would be def a solution.

is it on your radar? @aaronfrost @SanderElias

**Comment by SanderElias** [comment]
@stewwan Transferstate is implemented now. This is an opt-in mechanism. Aaron will bring out some samples soon

**Comment by stewwan** [comment]
that’s great! going to try this out on weekend.

**Comment by aaronfrost** [comment]
@stewwan try it and let us know.

**Comment by stewwan** [comment]
hey guys it works but there's a bug with getState. you can reproduce it by building `--prod`

```
    "@scullyio/ng-lib": "0.0.8",
    "@scullyio/scully": "0.0.53",
```

**Comment by stewwan** [comment]
![ezgif com-video-to-gif](https://user-images.githubusercontent.com/719763/72199456-163bca80-341b-11ea-87a3-46cfd7ab142d.gif)

**Comment by stewwan** [comment]
my workaround

```ts
  getState(key) {
    const SCULLY_SCRIPT_ID = `scully-transfer-state`;
    const SCULLY_STATE_START = `___SCULLY_STATE_START___`;
    const SCULLY_STATE_END = `___SCULLY_STATE_END___`;

    try {
      const el: any = document.getElementById(SCULLY_SCRIPT_ID);
      const html = el.text;
      let result = '';
      // Parse the scully state out of the next page
      const startIndex = html.indexOf(SCULLY_STATE_START);
      if (startIndex !== -1) {
        const afterStart = html.split(SCULLY_STATE_START)[1] || '';
        const middle = afterStart.split(SCULLY_STATE_END)[0] || '';
        result = middle;
      } else {
        result = null;
      }
      const newState = JSON.parse(unescapeHtml(result));
      return of(newState[key]);
    } catch (err) {}
  }
```

**Comment by SanderElias** [comment]
I fixed this issue in my in pr #182 
Will be available after it gets merged in.
I'm closing this issue now. If there are reamining issues with this, please open a new issue
